### PR TITLE
Add coverity and bandit checks

### PR DESCRIPTION
**Pull request by leshikus**
Subj for the https://scan.coverity.com/projects/mlir-extensions

### CODE REVIEW

File: .github/workflows/coverity.yml [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+            -DLLVM_ENABLE_ASSERTIONS=ON                                    \
+            -DLLVM_USE_LINKER=gold                                         \
+            -DLLVM_INSTALL_UTILS=ON                                        \
+            -DLLVM_TARGETS_TO_BUILD=X86                                    \
+            -DLLVM_ENABLE_BINDINGS=OFF                                     \
+            -DLLVM_ENABLE_ZSTD=OFF                                         \
+            -DCMAKE_INSTALL_PREFIX=/home/runner/work/llvm-mlir/_mlir_install
+          cmake --build _build --target install
+
+      - name: Build IMEX and run lit tests
```
Comment (Garra1980) [inline .github/workflows/coverity.yml:122]: this stage also does coverity build as far as I can see - line 132

Comment (leshikus) [reply .github/workflows/coverity.yml:122]: Thanks, renamed

**Comment by Garra1980** [comment]
Let's rename PR name

**Review by Garra1980 — APPROVED** [review]
LGTM, thanks a lot!

**Pull request merged**